test/integration/apimachinery/admission_test.go

```diff
@@ -0,0 +1,342 @@
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/deployments"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/daemonsets"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/replicasets"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/podsecuritypolicies"] = "true"
+		opts.APIEnablement.RuntimeConfig["extensions/v1beta1/networkpolicies"] = "true"
+	})
+	defer master.Cleanup()
+
+	if _, err := master.Client.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}); err != nil {
+		t.Fatal(err)
+	}
+
+	fail := admissionv1beta1.Fail
+	// Attaching Admission webhook to API server
+	_, err = master.Client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(&admissionv1beta1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{Name: "integration.test.k8s"},
+		Webhooks: []admissionv1beta1.Webhook{{
+			Name: "integration.test.k8s",
+			ClientConfig: admissionv1beta1.WebhookClientConfig{
+				URL:      &webhookEndpoint,
+				CABundle: localhostCert,
+			},
+			Rules: []admissionv1beta1.RuleWithOperations{{
+				Operations: []admissionv1beta1.OperationType{admissionv1beta1.OperationAll},
+				Rule:       admissionv1beta1.Rule{APIGroups: []string{"*"}, APIVersions: []string{"*"}, Resources: []string{"*/*"}},
+			}},
+			FailurePolicy: &fail,
+		}},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	client := master.Dynamic
+	testData := getAdmisssionTestData()
+
+	_, resources, err := master.Client.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		t.Fatalf("Failed to get ServerGroupsAndResources with error: %+v", err)
+	}
+	for _, list := range resources {
+		defaultGroupVersion, err := schema.ParseGroupVersion(list.GroupVersion)
+		if err != nil {
+			t.Errorf("Failed to get GroupVersion for: %+v", list)
+			continue
+		}
+		for _, resource := range list.APIResources {
+			t.Run(resource.Name, func(t *testing.T) {
+				kind := resource.Kind
+				if kindWhiteList.Has(kind) {
+					t.Skip("whitelisted")
+				}
+				verbs := resource.Verbs
+				ae := admissionElement{}
+				gvk := defaultGroupVersion.WithKind(kind)
+				ae.gvk.Group = gvk.Group
+				ae.gvk.Version = gvk.Version
+				ae.gvk.Kind = gvk.Kind
+				if len(resource.Version) > 0 {
+					ae.gvk.Version = resource.Version
+				}
+				if len(resource.Group) > 0 {
+					ae.gvk.Group = resource.Group
+				}
+				if strings.Contains(resource.Name, "/") {
+					ae.resource = strings.Split(resource.Name, "/")[0]
+					ae.sub = strings.Split(resource.Name, "/")[1]
+				} else {
+					ae.resource = resource.Name
+					ae.sub = ""
+				}
+				createObj, ok := testData[ae]
+				if !ok {
+					t.Logf("Object: %+v has no test data, skipping", ae)
+					t.Skip()
+				}
+				obj, err := createObj()
+				if err != nil {
+					t.Errorf("Failed to create Unstructured object with error: %+v", err)
+				}
+				for _, verb := range verbs {
```

Comment:
we'll probably want to do these in order, so we can exercise the lifecycle and not have to repeat ourselves:

if has(verbs,"create") {
  ... exercise create
  
  if has(verbs, "update") {
    ... update some label or annotation to "update"
  }
  if has(verbs, "patch") {
    ... patch some label or annotation
  }
  if has(verbs, "connect") {
    special-case stuff here TBD
  }
  if has(verbs, "delete") {
    delete
  }
  if has(verbs, "deletecollection") {
    create a new instance, then call deletecollection
  }
}

Reply:
for all the normal persistable resources (not subresources, and not whitelisted weird things like SubjectAccessReview or WebhookConfiguration types), the "create if needed" piece can grab the fixture data and POST it.

Reply:
rather than iterating over the verbs, put the verb tests in the order we want to run them and make them conditional on the resource supporting each verb